### PR TITLE
GUI Settings (market + core)

### PR DESCRIPTION
**Pull request by cis-shubham-t**

### CODE REVIEW

**Comment by AllienWorks** [comment]
Settings GFX preview: https://imgur.com/a/Z40rBNj

So far, I'm done. When you do your thing @cis-shubham-t, I will do the final tweaks afterwards ;)

**Comment by cis-shubham-t** [comment]
Awesome @AllienWorks 
Thanks for the update :)

**Comment by vikas-cis** [comment]
@zaSmilingIdiot @HenkSwardt please find the below sheet where everything has been added related to setting.

https://docs.google.com/spreadsheets/d/1Dit0CCHMkij7RsAV3ERMc1c8W8nQAHMd-OYhsFWgJJY/edit#gid=0

Thanks

**Comment by AllienWorks** [comment]
We have some outgoing links in the GUI (regarding various "read more" links + all of the links in Help section) - can we whitelist those? Currently the links are not doing anything (worked before), I suspect it's some kind of security measure..

**Links to be whitelisted:**
- https://particl.wiki/ (whole domain)
- https://particl.io/ (whole domain or just https://particl.io/user-guides)
- https://riot.im/app/#/group/+particl:matrix.org
- https://discordapp.com/invite/2tVJaZ9
- https://t.me/particlproject
- https://particl.news/
- https://twitter.com/particlProject
- https://www.facebook.com/ParticlProject/
- https://www.reddit.com/r/Particl/
- https://plus.google.com/+Particl
- https://mastodon.social/@particl
- https://www.youtube.com/c/Particl
- https://bitcointalk.org/index.php?topic=1835782.0
- https://github.com/particl
- https://www.transifex.com/particl/particl-website/

File: src/app/wallet/settings/settings.component.html [hunk 1]
```diff
@@ -444,7 +444,7 @@ <h4 class="option">Map port using UPnP</h4>
 
           <div class="item">
             <h4 class="option">Port forwarding</h4>
-            <p class="desc">Alternatively, you can configure your router to forward port <code>51738</code></p>
+            <p class="desc">Alternatively, you can configure your router to forward port<!-- <code>51738</code> --></p>
```
Comment (AllienWorks) [inline src/app/wallet/settings/settings.component.html:447]: Why did you hide this? How are users supposed to port forward, when they don't know the port? (The port doesn't change.)

Comment (zaSmilingIdiot) [reply src/app/wallet/settings/settings.component.html:447]: The port does change, depending on whether you are on testnet or mainnet :)
Also, while its not likely to be used outside of mainnet/testnet, the daemon can be configured with the 
`-port=<>`
flag, which does change the port value. This is also possible to be configured via the Desktop by passing it in as a flag when starting the Desktop client (or the daemon might have been configured via a particl.conf file containing the value).

**Comment by AllienWorks** [comment]
Closes #582

**Comment by zaSmilingIdiot** [comment]
The GUI here assumes that it is the source of truth for a variety of settings. 
This however, is not the general case: the particl.conf file can be created by a user individually, different datadir's for the deamon may be used, which allow for different daemon configuration, etc. Settings that relate to particl.conf changes, as well as those that read from the daemon need to be read in at the time of changing settings to ensure that the GUI has the latest values of those settings, rather than storing them and assuming that the stored values are the used values.

I've moved this to paused, as it is not priority at the moment.

**Comment by vikas-cis-zz** [comment]
Hye @zaSmilingIdiot iirc the settings did change the conf setting on the time of changing it from gui but yeah it will not restart the daemon at the same time, the conf files changes will be applicable once it has been restarted

Thanks

**Comment by zaSmilingIdiot** [comment]
True, but here are the more pressing concerns with the changes here:

1. Currently, writing to the particl.conf file overwrites whatever is in an existing particl.conf file. Since a user may have a custom particl.conf file, the GUI should not be overwriting this (deleting all other configuration specified in the file), but rather modifying any values if they are different, or otherwise creating them alongside the existing config.

2. Any configuration written to the particl.conf file is also currently stored in the GUI as well (along with the other settings the GUI allows to be configured). Where this is a problem is that the particl.conf file is never read back, so the GUI assumes that the changes it has stored itself (in localstorage) must be the correct settings. But this does not cater for the fact that the user may edit the particl.conf file themselves, or even allow for a different particl.conf file to be used. So when the GUI displays the Core settings to the user, the settings it displays is not necessarily what is currently in the configuration.

3. Extending point 3 above, there are a couple of settings that require interaction with the daemon. Again, since the user may set these values without using the GUI, we should be taking the action that when the user opens the settings page, the GUI query's the current values from the daemon and uses those as the current displayed values. There is no need to store those specific values.

I've already done some work in modifying the changes to fix up the above issues. Again, I'm just marking this as paused for now as other changes have priority at this point. I'll get back to working on it in the near future.

**Pull request closed without merging**